File: android/payment/app/src/main/kotlin/com/weekendinc/example/payment/DemoData.kt
package com.weekendinc.example.payment

import com.weekendinc.aipos.domain.entity.MerchantInfo
import com.weekendinc.aipos.domain.entity.Product
import com.weekendinc.aipos.domain.valueobject.Money
import com.weekendinc.aipos.domain.valueobject.ProductId

/**
 * Identitas merchant contoh.
 *
 * `profileId` dikosongkan supaya cocok dengan [MerchantInfo.DEFAULT_PROFILE_ID] — profil
 * lingkungan uji MineSec yang sepasang dengan lisensi `public-test.license` di `assets`.
 * Merchant sungguhan mengoper profil dan TID/MID miliknya sendiri.
 */
val demoMerchant = MerchantInfo(
    id = "merchant-demo",
    name = "Toko Contoh AIPos",
    address = "Jl. Contoh No. 1, Jakarta",
    terminalId = "TID00001",
    mid = "MID00001",
)

/** Satu produk contoh — cukup untuk mendemokan alur bayar tanpa katalog sungguhan. */
val demoProduct = Product(
    id = ProductId("demo-kopi-susu"),
    name = "Kopi Susu",
    price = Money.fromRupiah(25_000),
    barcode = "8990000000001",
    category = "Minuman",
    stock = 99,
)

File: ios/payment/Sources/DemoData.swift
import AIPosSDK
import Foundation

/// Identitas merchant contoh.
///
/// `profileId` diisi eksplisit dengan bawaan SDK: nilai bawaan parameter Kotlin tidak
/// ikut terbawa ke Swift, jadi konstruktor di sini menuntut argumen ini walau di Kotlin
/// ia opsional.
let demoMerchant = MerchantInfo(
    id: "merchant-demo",
    name: "Toko Contoh AIPos",
    address: "Jl. Contoh No. 1, Jakarta",
    terminalId: "TID00001",
    mid: "MID00001",
    profileId: MerchantInfo.companion.DEFAULT_PROFILE_ID
)

/// Satu produk contoh — cukup untuk mendemokan alur bayar tanpa katalog sungguhan.
///
/// Dirakit lewat `productOf`, bukan konstruktor `Product` langsung: `Money` dan
/// `ProductId` adalah value class Kotlin yang tidak terbawa ke Swift.
let demoProduct: Product = IosDomainFactoryKt.productOf(
    id: "demo-kopi-susu",
    name: "Kopi Susu",
    priceRupiah: 25_000,
    barcode: "8990000000001",
    category: "Minuman",
    stock: 99,
    description: "",
    imageUrl: ""
)
